app/domain/authentication/validate_security.rb

```diff
@@ -68,15 +68,15 @@ def validate_user_has_access
       end
 
       def user_role_id
-        @user_role_id ||= @role_class.roleid_from_username(@account, @user_id)
+        role_cache.roleid_from_username(@account, @user_id)
```

Comment:
In this class, too, we can do away with `MemoizedRole` and just use standard `||=` memoization.  Again, in the initializer, change it to `Role`, and then here change to:

```ruby
@user_role_id ||= @role_class.roleid_from_username(@account, @user_id)
```